packages/esm-patient-test-results-app/src/filter/filter-set.tsx

```diff
@@ -22,43 +23,74 @@ const FilterSet: React.FC<FilterSetProps> = ({ hideFilterSetHeader = false }) =>
   const { t } = useTranslation();
   const { resetTree } = useContext(FilterContext);
   const [searchTerm, setSearchTerm] = useState('');
+  const [treeDataFiltered, setTreeDataFiltered] = useState(roots);
   const [showSearchInput, setShowSearchInput] = useState(false);
 
+  useEffect(() => {
+    setTreeDataFiltered(roots);
+  }, [roots]);
+
+  function filterTreeNode(inputValue, treeNode) {
+    // If the tree node's display value contains the user input, or any of its children's display contains the user input, return true
+    if (
+      treeNode?.display.toLowerCase().includes(inputValue.toLowerCase()) ||
+      (treeNode?.subSets && treeNode.subSets.some((child) => filterTreeNode(inputValue, child)))
+    ) {
+      return true;
+    }
+
+    // Otherwise, return false
+    return false;
+  }
+
+  function handleInputChange(value) {
+    setSearchTerm(value);
+    const filteredData = roots.filter((node) => filterTreeNode(value, node));
+    setTreeDataFiltered(filteredData);
+  }
+
   return (
     <div className={!tablet ? styles.stickyFilterSet : ''}>
       {!hideFilterSetHeader &&
         (!showSearchInput ? (
           <div className={styles.filterSetHeader}>
             <h4>{t('tree', 'Tree')}</h4>
             <div className={styles.filterSetActions}>
-              <Button
-                kind="ghost"
-                size="sm"
-                onClick={resetTree}
-                renderIcon={(props) => <TreeViewAlt size={16} {...props} />}
-              >
-                {t('resetTreeText', 'Reset tree')}
-              </Button>
+              {searchTerm ? (
```

Comment:
I don't see why this needs the `searchTerm` guard? I.e., you should be able to click the reset button regardless of whether you've typed a `searchTerm`, right? Otherwise, how to I uncheck everything?